### ISSUE TITLE
支持 Ultralytics YOLO26（YOLOv26）：ONNX 导出适配、后处理与服务接入

### ISSUE THREAD

**Issue by lona-cn** [lona-cn/vision-simple#34]
## 实现与验收结果（已完成）

已在 main 提交并推送：[4c6e030 — feat: support YOLO26 detection segmentation pose and OBB](https://github.com/lona-cn/vision-simple/commit/4c6e030)。

- 新增 `kV26 = 26`，检测、分割、姿态、OBB 均支持 raw 与 NMS-free 输出；共享解码，严格验证元数据与形状，NMS-free 不二次抑制。
- 接入现有 C++、配置、流水线与 HTTP 接口；YOLOv10/YOLO11 与服务批次／错误语义保持不变。
- 新增 [可复现导出脚本](https://github.com/lona-cn/vision-simple/blob/4c6e030/scripts/export_yolo26.py)，固定 Ultralytics 8.4.159、opset 17、静态 batch=1，记录权重与 ONNX SHA256、参数及张量元数据；CPU FP16 转换后拓扑排序并执行 checker/ORT 加载验证。
- [中文说明及兼容矩阵](https://github.com/lona-cn/vision-simple/blob/4c6e030/README.md) 与英文 README 已更新。

### 实际验证范围

环境：Windows x64 Release，C++ ORT 1.20.0 / DirectML 1.15.4；参考端 Python ORT 1.24.3。

| EP | raw FP32 | NMS-free FP32 | raw FP16 | NMS-free FP16 |
|---|---|---|---|---|
| CPU | 四任务通过 | 四任务通过 | 四任务通过 | 四任务通过 |
| DirectML | 四任务对照通过 | 四任务对照通过 | 四任务推理通过 | 四任务 ORT 初始化失败 |
| CUDA / TensorRT | 未验证 | 未验证 | 未验证 | 未验证 |

CPU 共 16 个真实 nano 导出组合通过 C++ Run 与 HTTP 验证，覆盖方形／宽图；DML 8 个 FP32 组合通过几何对照。FP32 框误差 <0.5 px、关键点／OBB 角点误差 <0.001 px；CPU FP16 最大框误差 0.75 px、关键点误差 0.375 px、置信度绝对误差 <0.007；分割对照最小 mask IoU 0.941。

OBB raw 保持本项目多边形 IoU NMS（不是 Ultralytics 概率 IoU），独立 oracle 已核对；同一方图分别保留 180 与 175 个框。关键点不裁剪图外坐标，mask 在 logits 插值后阈值化，Letterbox 为黑色 padding；对照时已显式处理这些既有契约差异。

一次 CPU 方图阶段记录（预处理／推理／后处理，ms）：检测 raw FP32 1.381 / 25.476 / 0.489；NMS-free FP32 1.200 / 21.912 / 0.010。仅为功能烟测，不是性能保证。

### 已通过的构建与回归

- `xmake build -y server`
- `test_yolo_postprocess`：旧版本、两种 YOLO26 布局、歧义 shape、阈值／NMS、非法输出和元数据。
- `test_yolo_tasks`：三任务 FP32/FP16、mask／关键点／旋转框、pipeline、非法契约。
- `test_infer_inputs`：输入边界、恢复、YOLO26 加载元数据及相同 [1,6,6] 布局判别。
- `scripts/test_yolo_tasks_http.py`：YOLO26 两种模式、FP16、序列化及现有协议／生命周期回归。
- `scripts/test_http_regression.py`：原有 YOLO/OCR 正确性、批次顺序、失败恢复、并发、卸载、超时／容量与启动错误回归。

### 部署限制与迁移

**当前 DML 栈不支持本次 NMS-free FP16 产物**；请选择 FP32 或已验证的 raw FP16，失败返回受控 `model_load_failed`，无静默模式替换。CUDA/TensorRT 未验证。此限制已按原验收要求记录，不表示所有 EP 兼容。

`InferYOLOTask::Create` 新增显式版本参数；旧调用在 task 后补 `YOLOVersion::kV11`，YOLO26 使用 `YOLOVersion::kV26`，可选 device_id 顺延。模型权重／构建产物不提交；提交包含微型回归 fixtures。

---

## 目标

为 vision-simple 增加明确、可验证的 Ultralytics **YOLO26**（常写作 YOLOv26）支持，复用现有 ONNXRuntime、图像预处理、推理流水线及服务接口，而不是要求用户将新模型伪装成 `kV10` / `kV11`。实现已完成，验收结果与部署限制见上方。

关联：#3 已完成 YOLO11 的 det / seg / pose / obb；YOLO26 需要独立验证导出契约。

## 实施前的实现与差距（历史背景）

- `app/source/runtime/infer/Infer.h`：`YOLOVersion` 当前只有 `kVCustom`、`kV10`、`kV11`，没有 YOLO26 标识。
- `app/source/runtime/infer/private/InferYOLO.cpp`：`ValidOutputShape` 和 `YOLOFilter::operator()` 只处理 v10 / v11。v10 接受 `[1,K,6]`，解析 `xyxy + confidence + class_id`，不执行 NMS；v11 接受 `[1,4+nc,A]`，解析中心点／宽高与类别分数，然后执行按类别 NMS。
- 现有检测加载要求单输入、单输出、静态 `[1,3,H,W]`、FP32/FP16 张量及有效的 `names` 元数据；工作区按已知 shape 分配。新增版本不应顺带放宽为任意动态 shape 或 batch。
- `app/source/programs/server/private/TaskRegistry.cpp`：检测通过 `magic_enum::enum_cast<YOLOVersion>` 解析配置版本；seg / pose / obb 加载目前显式要求 `kV11`，不能仅开放版本字符串就视为兼容。

## YOLO26 导出契约

按当前 Ultralytics 官方文档，检测模型有两条路径；导出器版本必须固定并记录，不能依赖不同版本的默认行为：

| 导出模式 | 检测输出（batch=1） | 本项目需要的后处理 |
|---|---|---|
| 原始 one-to-many（当前文档 `nms=None`） | `[1,4+nc,A]`，框为 `xywh` | 阈值过滤、坐标还原、按类别 NMS |
| NMS-free one-to-one（当前文档 `nms=False`） | `[1,K,6]`，每行为 `x1,y1,x2,y2,score,class_id` | 阈值过滤、坐标还原；**不得再次执行 NMS** |

`A` 随输入大小变化；`K` 受导出时的检测上限等因素影响，不应写死为 8400 / 300。`nms=True` 的内嵌 NMS 导出也可能具有 `[1,K,6]`，因此 shape 只能说明布局，不能证明模型采用了 NMS-free 检测头。内嵌 NMS 导出需独立确认，未经验证不得列入支持矩阵。

### 1. 固定模型与导出基线

- [x] 固定 Ultralytics 版本，记录权重来源／校验值、ONNX opset、导出参数、ORT 版本及 EP；分别生成检测原始输出与 NMS-free 输出的真实 ONNX 产物。
- [x] 以 `yolo26n.pt`、batch=1、静态输入、FP32 建立基线，再验证 FP16；记录实际输入／输出 dtype、shape、`names` 和任务元数据。不得假定 FP16 权重意味着全部输出均为 FP16。
- [x] 提供与固定导出器版本一致的完整导出命令；以产物和对照推理验证参数含义，而非直接把在线文档示例当作已验证命令。

### 2. 检测运行时适配

- [x] 新增 `YOLOVersion::kV26 = 26`，保持现有枚举值不变，更新版本校验与分派。
- [x] 将现有 v10 end-to-end 解码和 v11 原始输出解码整理成可共享的内部实现，v26 按经验证的输出契约选择；避免复制整套后处理，也不将 v26 偷换为 v10 / v11。
- [x] 结合 task／导出元数据与严格 shape 校验确定支持格式；元数据不足、契约冲突或布局歧义时明确拒绝，不按文件名猜测。尤其要覆盖 `nc=2`、候选数为 6 时可能出现的布局歧义。
- [x] 沿用 Letterbox 与坐标逆变换，校验输出长度、有限数值、类别索引和类别元数据；无目标返回空结果，非法模型通过 `VSResult` 返回受控错误。
- [x] NMS-free 路径不执行额外 NMS；原始输出路径保留按类别 NMS。明确阈值边界语义，避免共享解码重构改变既有 v10 / v11 行为。

### 3. 配置与服务接入

- [x] canonical 模型配置接受 `task: yolo` / `version: kV26`，经现有 `TaskRegistry`、模型租约及流水线加载与执行。
- [x] 保持现有 C++ 检测结果结构、`POST /v0/infer/yolo` 与 `POST /v1/infer/yolo` 的成功响应和错误语义，不为新版本新增专用 HTTP 路由。
- [x] README 补充支持矩阵、导出步骤、精度／shape 限制及以下配置示例（现已可用；权重需自行导出）：

```yaml
models:
  - task: yolo
    name: yolo26n
    version: kV26
    files:
      model: assets/yolo26n.onnx
```

### 4. 分割、姿态与旋转框

- [x] 分别核对 YOLO26 seg / pose / obb 的原始输出及 end-to-end 输出，确认 mask prototype／系数、关键点布局、旋转框角度和坐标语义，再适配各任务解码器与注册表。
- [x] 不把“检测支持”宣传成全任务支持，也不把 YOLO11 解码器直接套到 YOLO26。各任务分别提交真实模型对照结果，再开放其 `kV26` 配置。

分类、语义分割、深度估计、YOLOE、训练、动态输入和 batch>1 不由本 issue 默认扩展；本 issue 跟踪现有 det / seg / pose / obb 任务的 YOLO26 适配。

## 验收标准

- [x] 真实 YOLO26 ONNX 模型可通过 C++ API 与 HTTP 服务运行；检测两种输出模式分别与相同导出设置、预处理及阈值的 Ultralytics 参考结果对照，记录类别、置信度、框坐标容差。
- [x] 覆盖空检测、自定义类别数、非方形图像／Letterbox padding、不同检测上限、非法 shape、NaN/Inf、越界／非整数 class_id 及缺失元数据。
- [x] 用高度重叠候选验证：end-to-end 输出不会被二次 NMS 删除；原始输出仍正确执行按类别抑制。
- [x] 扩展现有 `test_yolo_postprocess.cpp`、输入校验与各任务测试中的行为回归；v10 / v11 及既有服务批次顺序、整批失败语义不回退。
- [x] seg / pose / obb 分别完成 mask、关键点与旋转框对照；仅检测通过时不得关闭全部任务验收。
- [x] CPU EP 为必验基线；DML / CUDA / TensorRT 按实际可用环境逐项记录结果与算子限制，未测 EP 明确标注，不能因 ORT 可加载就宣称全部后端兼容。记录预处理／推理／后处理耗时，不直接套用官方加速比例。

## 参考

- [Ultralytics YOLO26](https://docs.ultralytics.com/models/yolo26/)
- [End-to-End Detection：检测头、输出格式与导出兼容性](https://docs.ultralytics.com/guides/end2end-detection/)

上述验收已完成；实际支持范围以本 issue 顶部验收矩阵与 README 为准。
